File: src/main/resources/static/app/rules/sensorrules/entities/AlarmSensorRule.js
class AlarmSensorRule extends SensorRule{
		
	
	constructor(sensorMeasureThresholdSettings, id, enabled, timeBetweenTriggers, notificationType) {
		super(sensorMeasureThresholdSettings, id, enabled, timeBetweenTriggers, notificationType);
	}
	
	isValid() {
		return super.isValid();
	}
	
	accept(sensorRuleVisitor) {
		sensorRuleVisitor.visitAlarmSensorRule(this);
	}
	
	static get alarmSensorRuleType() {
    	return 'Sensor alarm';
  	}

}

File: src/main/java/com/iotassistant/models/rules/AlarmSensorRule.java
package com.iotassistant.models.rules;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;

import com.iotassistant.models.SensorMeasureThresholdSettings;
import com.iotassistant.models.notifications.NotificationTypeEnum;

@Entity
@DiscriminatorValue("alarmSensorRule")
public class AlarmSensorRule extends SensorRule{


	public AlarmSensorRule() {
		super();
	}
	
	public AlarmSensorRule(SensorMeasureThresholdSettings sensorMeasureThresholdSettings, NotificationTypeEnum notificationType, 
			RuleTriggerIntervalEnum timeBetweenTriggers,  boolean enabled) {
		super(sensorMeasureThresholdSettings, enabled, timeBetweenTriggers, notificationType);
	}
	


	public Integer getId() {
		return id;
	}


	public void setId(Integer id) {
		this.id = id;
	}
	

	@Override
	public void accept(SensorRuleVisitor sensorRuleVisitor) {
		sensorRuleVisitor.visit(this);		
	}
	


}
